docs/guides/end-to-end-testing/testing-strategies/azure-active-directory-authentication.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: Azure Active Directory Authentication
+slug: /guides/end-to-end-testing/azure-active-directory-authentication
+---
+
+:::info
+
+## <Icon name="graduation-cap"></Icon> What you'll learn
+
+- Log in to
+  [Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis)
+  through the UI with [`cy.origin()`](/api/commands/origin)
+
+:::
+
+:::caution
+
+This guide is designed for testing against a Single Page Application (SPA) that
+uses
+[Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis)
+to authenticate users. For this guide, the Microsoft Authentication Library
+[`@azure/msal-browser`](https://www.npmjs.com/package/@azure/msal-browser)
+package is used by the webapp to broker this authentication.
+
+This guide can also serve as a foundation for testing other webapps with Cypress
+that use Azure Active Directory services with other frameworks, such as React,
+Angular, or Vue.
+
+:::
+
+## Microsoft AAD Application Setup
+
+For this guide, we are mainly going to focus on setting up Cypress to test
+against an Azure Active Directory (AAD) Webapp. Please clone the
+[Microsoft Identity Javascript Tutorial](https://github.com/Azure-Samples/ms-identity-javascript-tutorial/tree/c1956b658efa331bb5df11a0038ad32d12dad3ce/1-Authentication/1-sign-in)
+and follow the steps to set up your application.
+
+Once set up, you will need to do a few things to the `index.html` file:
+
+- Remove any integrity attributes inside `<script>` tags. This is to prevent any
+  SRI attribute mismatches, which can happen since Cypress must rewrite any
+  [framebusting code](/guides/references/configuration#modifyObstructiveCode).
+  This is recommended only when an application is under test.
```

Comment:
If we remove these while testing, does this impact the integrity of the e2e tests? is it possible to leave these integrity attributes?

Reply:
When you say "impact the integrity of the e2e test" are you referring that it should throw an error if the integrity is there or that users don't have a solid test because they removed the integrity? Or none of those thing? 😅 

Right now I don't think there is a way to keep the integrity values in the app `index.html` since we don't strip integrity for first party attributes:

Heres the code in the proxy  in `response-middleware` that enforces this
![](https://user-images.githubusercontent.com/3980464/212986925-b729d357-a120-4c19-b8dc-e5aa90e5087e.png)

I feel like this is something users would be unwilling to do, and a better option we could provide is have a unique flag for SRI stripping and when its turned on it strips SRI for everything. Then the user wouldn't have to modify their index file here and Cypress would just automatically do it on opt in.

Reply:
yeah the first statement